### PR TITLE
Added GitHub actions for CI

### PR DESCRIPTION
**Pull request by samkhawase**

### CODE REVIEW

**Comment by samkhawase** [comment]
@suhaibroomy I'm getting this error when I try to [build](https://github.com/navgurukul/soni-sori/pull/115/checks?check_run_id=1856167638) the app using gradle. I'm pretty sure that's because of Vikram's changes but I'm not proficient enough to resolve the .aar issue. Would you guys mind to take a look at this? 

```
FAILURE: Build failed with an exception.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.

Use '--warning-mode all' to show the individual deprecation warnings.
* What went wrong:
See https://docs.gradle.org/6.5/userguide/command_line_interface.html#sec:command_line_warnings
Execution failed for task ':chat:bundleDebugAar'.
326 actionable tasks: 97 executed, 229 up-to-date
> Direct local .aar file dependencies are not supported when building an AAR.
The resulting AAR would be broken because the classes and Android resources from any local .aar file dependencies would not be packaged in the resulting AAR. 
Previous versions of the Android Gradle Plugin produce broken AARs in this case too (despite not throwing this error). 
The following direct local .aar file dependencies of the :chat project caused this error: 
/home/runner/work/soni-sori/soni-sori/chat/lib/matrix-sdk-android-release.aar,
 /home/runner/work/soni-sori/soni-sori/chat/lib/matrix-sdk-android-rx-release.aar
```

**Comment by abhishekgupta92** [comment]
Shouldn't we also have one for develop branch?

**Comment by samkhawase** [comment]
> Shouldn't we also have one for develop branch?

@abhishekgupta92 We're using GitHub flow where we branch off of `main` and merge it back to it. 
The `develop` branch model is a different style called **GitFlow** which has `develop`, `feature`, `hotfix` and `release` branches.

File: .gitignore [hunk 1]
```diff
@@ -4,7 +4,6 @@ app/release
 *patch
 *.iml
 .gradle
-/local.properties
```
Comment (suhaibroomy) [inline .gitignore:7]: local.properties shouldn't be commited to git. Its mean't to be local only

File: local.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sdk.dir=ANDROID_SDK_ROOT
```
Comment (suhaibroomy) [inline local.properties:1]: shouldn't be committed to git, should be generated dynamically on CI if required

**Review by suhaibroomy — CHANGES_REQUESTED** [review]

File: local.properties (hunk 2 above)
Comment (samkhawase) [reply local.properties:1]: I didn't find any example to generate it dynamically, do you know how it can be done?

Comment (samkhawase) [reply local.properties]: Here's the error I get on `app:lint` :
```
Use '--warning-mode all' to show the individual deprecation warnings.
* What went wrong:
A problem occurred configuring project ':playground'.
> /home/runner/work/soni-sori/soni-sori/local.properties (No such file or directory)
```

File: .gitignore (hunk 1 above)
Comment (samkhawase) [reply .gitignore]: Fixed with latest commit.

**Review by samkhawase — COMMENTED** [review]
@suhaibroomy I've updated the PR and updated it with new changes. Can you please approve the changes?

**Pull request merged**